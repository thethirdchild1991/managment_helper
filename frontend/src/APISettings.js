export const APIAddr = 'http://localhost:5000'