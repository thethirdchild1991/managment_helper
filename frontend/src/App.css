/* * {
  padding: 0;
  margin: 0;
}
html, body, #root{
  height: 100%;
  width: 100%;
}
.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
} */

/* .App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  height: 200px;
} */
/* 
#mainLogo {
  
}

#logInForm{
  background-color: coral;
  text-align: center;
} */
/* .button{  
  display: block;
  width: 100%;
  border: none; 
  background-color: #4CAF50;
  padding: 10px 28px;
   font-size: 16px;
  cursor: pointer;
  text-align: center;  
} */
/* 
.App-link {
  color: #61dafb;
}
#footer{
  background-color: #282c34;
  text-align: center;
  position:absolute; 
  bottom:0; 
  width:100%; 
}
#nav{
  background-color: #61dafb;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}
main {
  background-color: aquamarine;
	min-height: 100%;
	display: grid;
	grid-template-rows: auto 1fr auto;
	grid-template-columns: 100%;
} */
/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */


* {
  margin: 0;
}

body {
  color: white;
}

#root {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

/* #root > * {
  padding: 20px;
} */

header, footer, nav, ul{  
  display: flex;
  flex-direction: row;
  /* align-items: center; */
  justify-content: center;  
}
header{
  background: #592E83;
  padding: 10px 0;
}
nav {
  background-color: blueviolet;
  color: white;
}

nav a{
  color: white;
}

li {
  display: block;
  padding: 0 10px;
}
main {
  display: flex;
  flex-grow: 1;
  flex-direction: row;
  justify-content: center;  
  background: #331E4D;
}

.authWrapper{
  display: flex;
  flex-direction: column;
  height: max-content;
}

.authTypeSelector{
  display: flex;
  flex-direction: row;
  background-color: blue;
}
.authTypeSelector div{
  display: flex;
  flex-grow: 1;
  justify-content: center;  
}

.authTypeSelector .active{
  background-color: green;
}

.authTypeSelector .notActive{
  background-color: red;
}

.signIn, .signUp{
  height: max-content;
  padding: 0 10px;
}
#signIn div, #signUp div{
  display: flex;
  flex-direction: column;
  padding: 10px 0;
}

#signIn .submitButton, #signUp .submitButton{
  background-color: aquamarine;
}

.submitButton{
  line-height: 4;
  width: 100%;
  margin: 10px 0;
}

.signUp{
  background-color: brown;
  
}
.signIn{
  background-color: tomato;
}

.hidden{
  display: none;
}

.projectTable{
  color: white;
  background-color: green;
}

footer {
  background: #9984D4;
  padding: 10px 0;
}
footer a{
  padding: 0 5px;
}

#NOPE{
  color: black;
}
