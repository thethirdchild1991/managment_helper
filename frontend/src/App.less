@empty-body-color: white;
@header-background-color : #592E83;
@main-background-color: #331E4D;;
@nav-background-color: blueviolet;

@mainWrapper-background-color: bisque;;
@mainWrapper-a-hover-color: green;

@authTypeSelector-active-color: green;
@authTypeSelector-notActive-color: red;
@signUp-background-color: brown;
@signIn-background-color: tomato;

@projectTable-elem-hover-color: red;
@formWrapper-input-focus-background-color:#9984D4;



* {
    margin: 0;
    padding: 0;
    border: 0;
}
  
body {
    color: @empty-body-color;
}

p{
    color: black;
}
  
#root {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
}
    
header, footer, #nav, #nav ul{  
display: flex;
flex-direction: row;
/* align-items: center; */
justify-content: center;  
}
header{
    background: @header-background-color;
    padding: 10px 0;
}
nav {
    background-color: @nav-background-color;
    color: white;
    & a{
        color: white;        
    }
}


li {
display: block;
padding: 0 10px;
}
main {
display: flex;
flex-grow: 1;
flex-direction: row;
justify-content: center;      
background: @main-background-color;
}

.main {
display: flex;
flex-direction: column;    
width: max-content;  

}

.MainNavWrapper {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    flex-basis: 0;
    & nav{
        display: flex;
        flex-direction: column;
        flex-grow: 1;
        flex-basis: 0;
        background-color: @mainWrapper-background-color;
        color: black;
    }
    & a{
        background-color: @mainWrapper-background-color;
        color: black;
        display: inline-block;
        width: 100%;
        &:hover{
            background-color: @mainWrapper-a-hover-color;
        }
    }
    & li{
        padding: 0;
    }
}

.formWrapper{
    display: flex;
    flex-direction: column;    
    justify-content: center;
    & div{
        display: flex;
        justify-content: center;  
        flex-direction: row;
    }
}

#ProjectForm div, #CreateDeveloper div, #EditUser div{
justify-content: space-between;
}

.authWrapper{
display: flex;
flex-direction: column;
height: max-content;
}

.authTypeSelector{
    display: flex;
    flex-direction: row;   
    & div{
        display: flex;
        flex-grow: 1;
        justify-content: center;          
    }
    & .active{
        background-color: @authTypeSelector-active-color;
    }
    & .notActive{
        background-color: @authTypeSelector-notActive-color;
    }
}

.signIn, .signUp{
height: max-content;
padding: 0 10px;
}
#signIn div, #signUp div{
display: flex;
flex-direction: column;
padding: 10px 0;
}

#signIn .submitButton, #signUp .submitButton{
    background-color: aquamarine;
}

.submitButton{
    line-height: 4;
    width: 100%;
    margin: 10px 0;
}

.signUp{
    background-color: @signUp-background-color;
}
.signIn{
    background-color: @signIn-background-color;
}

.hidden{
    display: none;
}

.projectTable{
    color: white;
    background-color: green;
    width: max-content;
    border: solid 3px black;
    border-collapse: collapse;
    align-self: center;
    & th,td{
        border: solid 2px black;        
    }
    & tr{
        &:hover{
            background-color: @projectTable-elem-hover-color;
        }
    }

    & a{
        display: inline-block;
        width: 100%;
        &:hover{
            background-color:@projectTable-elem-hover-color;
        }
    }

}


.ProjectEditWrapper {
    display: flex;
    flex-direction: row;
    & input{
        flex-grow: 1;
        border: solid 3px black;
        line-height: 3;
    }
}


.formWrapper input:focus{
    background-color: @formWrapper-input-focus-background-color;
}

footer {
    background: #9984D4;
    padding: 10px 0;
    & a{
        padding: 0 5px;    
    }
}

#NOPE{
    color:black;
}
