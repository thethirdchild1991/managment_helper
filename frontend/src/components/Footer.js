import React, {Component} from 'react'
import {footerSocNetworksData} from '../configs/socNetSettings'


const footerSocNetworksElements = footerSocNetworksData.map( value => {
    return (
        <a href={value.link} target="blank" key={value.link}>        
            <img alt={value.alt} src={value.img} />
        </a>
        )
})
class Footer extends Component{
    render () {
        return(
            <footer id='footer'>
            {footerSocNetworksElements}
            </footer>
        );
    }
}

export default Footer;