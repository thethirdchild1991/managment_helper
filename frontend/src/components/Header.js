import React from 'react'
import {Component} from 'react'
import MainLogo from './MainLogo'

class Header extends Component{
    render(){
        return (
            <header className = 'header'>
                <MainLogo />
            </header>        
        );
    }    
}

export default Header;