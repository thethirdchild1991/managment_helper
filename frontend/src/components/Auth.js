import React, {Component} from 'react'
import {
    Redirect,    
  } from "react-router-dom";
import AppForm from './AppForm'
import SignInFormConfig from '../configs/SignInFormConfig'
import SignUpFormConfig from '../configs/SignUpFormConfig'
import {API} from '../configs/APISettings'


class Auth extends Component{
    constructor(props){
        super(props)        
        this.authHandler = props.authHandler
        this.state = {
            authState : props.authState,            
            active : 'signInTab'
        }
    }

    onClickHandler = event => {
        this.setState({
            active : event.target.id
        }, () =>{console.log(this.state)})
    }

    onSignInHandler = param =>{ 
        const AuthStatus = JSON.parse(param.loggedIn)                  
        if(AuthStatus === true ){
            localStorage.setItem('loggedIn', AuthStatus)            
            this.setState({authState : AuthStatus})
        }else{
            localStorage.setItem('loggedIn', false)            
            this.setState({authState : false})
        }        
    }

    onSignUpHandler = event =>{        
    }

    

    render(){       
        
        if(this.state.authState === true){            
            console.log('redirect')
            return <Redirect to='/index'/>            
        }        
                      
        return(
        <main id='main'>
            <div className='authWrapper'>
                <div className='authTypeSelector'>
                    <div 
                        className={this.state.active === 'signInTab' ? 'active' : 'notActive'} 
                        id='signInTab' 
                        onClick={this.onClickHandler}>
                            SignIn
                    </div>
                    <div 
                        className={this.state.active === 'signUpTab' ? 'active' : 'notActive'} 
                        id='signUpTab'
                        onClick={this.onClickHandler}>
                            SignUp
                    </div>
                </div>
            <AppForm 
                id='signIn'
                mainClassName='signIn'
                additionalClassName={this.state.active === 'signInTab' ? 'notHidden' : 'hidden'}
                proto={SignInFormConfig}
                submitText='SignIn'
                url={API.singIn}
                extSubmitHandler={this.onSignInHandler}                
            />   
                
            <AppForm 
                id='signUp'
                mainClassName='signUp'
                additionalClassName={this.state.active === 'signUpTab' ? 'notHidden' : 'hidden'}
                proto={SignUpFormConfig}
                submitText='SignUp'
                url={API.singUp}
                extSubmitHandler={this.onSignUpHandler}
            /> 
            </div>
            
        </main>
        )        
    }
}

export default Auth