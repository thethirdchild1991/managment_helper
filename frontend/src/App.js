import React, { Component } from 'react';
import Header from './Header'
import Footer from './Footer';
import Nav from './Nav'
import Main from "./Main";
import './App.css';


class App extends Component {
  render() {
    return (
      <>
        {/* <Header />   */}
        {/* <Nav /> */}
        {/* <Main />   */}
        <Footer />           
      </>      
    );
  }
}

export default App;
