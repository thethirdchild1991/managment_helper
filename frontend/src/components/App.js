import React, { Component } from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'
import {API} from '../configs/APISettings'
import Header from './Header'
import Footer from './Footer';
import Nav from './Nav'
import Main from "./Main";
import NOPE from './NOPE'
import Auth from './Auth'
import ProjectView from './ProjectView'
import editUserFormConfig from '../configs/editUserFormConfig'
import editProjectFormConfig from '../configs/editProjectFormConfig'

import '../App.css';

class App extends Component {
  constructor(props){
    super(props)  
    const loggedState = Boolean(localStorage.getItem('loggedIn')) 

    this.state = {
      authState : loggedState ? loggedState : false,
    }    
        
  }

  render() {
    return (
      <Router>
        <Header />
        <Switch>
          <Route 
            exact 
            path='/' 
            render={ () => <><Nav /><Main /></> }
          />
          <Route 
            path='/index' 
            render={ () => <><Nav /><Main /></> }
          />
          <Route 
            path='/auth' 
            component={Auth} 
          />
          <Route 
            exact 
            path='/project/:id' 
            render = { (param) => <ProjectView 
                                    data={param} 
                                    path={{
                                            fetch : API.selectProject,
                                            edit : API.updateProject,
                                            delete : API.delenpmteProject,
                                          }}
                                    proto={editProjectFormConfig}
                                  /> }              
          />
          <Route 
            exact 
            path='/user/:id' 
            render = { (param) => <ProjectView 
                                    data={param} 
                                    path={{
                                            fetch : API.selectUser,
                                            edit : API.updateUser,
                                            delete : API.deleteUser,
                                          }}
                                    proto={editUserFormConfig}
                                    /> }   
          />
          <Route component={NOPE} />
        </Switch>
        <Footer />
      </Router>
    );
  }
}

export default App;




