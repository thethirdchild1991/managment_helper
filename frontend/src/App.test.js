import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {APIAddr} from './configs/APISettings'


import App from './components/App'
import Header from './components/Header'
import Footer from './components/Footer'
import Nav from './components/Nav'
import Main from './components/Main'
import Auth from './components/Auth'
import ProjectView from './components/ProjectView'
import NOPE from './components/NOPE'

//-------------------------------------------------------------------------
describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('Nav', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <Nav />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('Main', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <Main />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('Auth', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <Auth />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('ProjectView', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <ProjectView />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('NOPE', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <NOPE />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
describe('Footer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( <Footer />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})
//-------------------------------------------------------------------------
// describe('LabeledInput', () => {
//   const LabeledInputParams = {
//     text:'agent',        
//     id: 'pro_agent',
//     required: true,
//     BR : true,

//   }

//   it('renders without crashing', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(                
//       <LabeledInput params={LabeledInputParams} />
//     , div);
//     ReactDOM.unmountComponentAtNode(div);
//   });
// });
// //-------------------------------------------------------------------------
// describe('AppForm', () => {
//   it('renders without crashing', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(                
//       <AppForm
//         id='SignUpForm'
//         url={APIAddr}
//         proto={SignUpFormConfig} 
//         submitText="SignUp" /> 
//     , div);
//     ReactDOM.unmountComponentAtNode(div);
//   });

  // test('has a valid snapshot', () => {
  //   const component = renderer.create( <App /> );
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });



// });