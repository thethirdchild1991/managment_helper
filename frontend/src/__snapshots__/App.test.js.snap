// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<footer
  id="footer"
>
  <a
    href="http://youtube.com"
    target="blank"
  >
    <img
      alt="youtube"
      src="http://ardeonova.com/sites/all/themes/riftek/img/videos.png"
    />
  </a>
  <a
    href="http://facebook.com"
    target="blank"
  >
    <img
      alt="facebook"
      src="http://ardeonova.com/sites/all/themes/riftek/img/logotype42.png"
    />
  </a>
  <a
    href="http://linkedin.com"
    target="blank"
  >
    <img
      alt="linkedin"
      src="http://ardeonova.com/sites/all/themes/riftek/img/socialnetwork168.png"
    />
  </a>
</footer>
`;
